tiledb-sys/src/datatype.rs

```diff
@@ -17,6 +18,7 @@ extern "C" {
     pub fn tiledb_datatype_size(type_: u32) -> u64;
 }
 
+#[allow(non_snake_case)]
```

Comment:
It's a bigger change but I kind of think this should change to be more similar to the smaller enums I've worked through - i.e. the `sys` library just has the constants and then we put the enum in the main crate.  We shouldn't need this.

Reply:
This one I'm not even sure why it started complaining when it was there before the PR. The linter disable was purely to tell clippy to quiet up so I didn't have to include tugging on this loose end.

Reply:
Honestly, I've got no idea why clippy suddenly started yelling about this. I just did this to silence the warning so I didn't have to tug on the string to fix it in the context of this PR.

Reply:
Does clippy run on all directories?  If not maybe that's why it didn't catch this.